dataikuapi/dss/ml.py

```diff
@@ -315,43 +313,979 @@ def save(self):
+        self.min_samples_leaf = self._register_numerical_hyperparameter("min_samples_leaf")
+        self.max_tree_depth = self._register_numerical_hyperparameter("max_tree_depth")
+        self.max_feature_prop = self._register_numerical_hyperparameter("max_feature_prop")
+        self.max_features = self._register_numerical_hyperparameter("max_features")
+        self.n_jobs = self._register_single_value_hyperparameter("n_jobs", accepted_types=[int])
+        self.selection_mode = self._register_single_category_hyperparameter("selection_mode", accepted_values=["auto", "sqrt", "log2", "number", "prop"])
+
+
+class XGBoostSettings(PredictionAlgorithmSettings):
+
+    def __init__(self, raw_settings, hyperparameter_search_params):
+        super(XGBoostSettings, self).__init__(raw_settings, hyperparameter_search_params)
+        self.max_depth = self._register_numerical_hyperparameter("max_depth")
+        self.learning_rate = self._register_numerical_hyperparameter("learning_rate")
+        self.gamma = self._register_numerical_hyperparameter("gamma")
+        self.min_child_weight = self._register_numerical_hyperparameter("min_child_weight")
+        self.max_delta_step = self._register_numerical_hyperparameter("max_delta_step")
+        self.subsample = self._register_numerical_hyperparameter("subsample")
+        self.colsample_bytree = self._register_numerical_hyperparameter("colsample_bytree")
+        self.colsample_bylevel = self._register_numerical_hyperparameter("colsample_bylevel")
+        self.alpha = self._register_numerical_hyperparameter("alpha")
+        self.lambda_ = self._register_numerical_hyperparameter("lambda")
+        self.booster = self._register_categorical_hyperparameter("booster")
+        self.objective = self._register_categorical_hyperparameter("objective")
+        self.n_estimators = self._register_single_value_hyperparameter("n_estimators", accepted_types=[int])
+        self.nthread = self._register_single_value_hyperparameter("nthread", accepted_types=[int])
+        self.scale_pos_weight = self._register_single_value_hyperparameter("scale_pos_weight", accepted_types=[int, float])
+        self.base_score = self._register_single_value_hyperparameter("base_score", accepted_types=[int, float])
+        self.impute_missing = self._register_single_value_hyperparameter("impute_missing", accepted_types=[bool])
+        self.missing = self._register_single_value_hyperparameter("missing", accepted_types=[int, float])
+        self.cpu_tree_method = self._register_single_category_hyperparameter("cpu_tree_method", accepted_values=["auto", "exact", "approx", "hist"])
+        self.gpu_tree_method = self._register_single_category_hyperparameter("gpu_tree_method", accepted_values=["gpu_exact", "gpu_hist"])
+        self.enable_cuda = self._register_single_value_hyperparameter("enable_cuda", accepted_types=[bool])
+        self.seed = self._register_single_value_hyperparameter("seed", accepted_types=[int])
```

Comment:
For `nthread` is not a hyperparameter and will never be searchable (as it has theoretically no influence on results), so unlike other single-valued hyperparameters, it should probably be a simple gettable + settable property, meaning getting it should return the direct value instead of a "single-valued hyperparameter" object.

It is probably the same for:
- `n_jobs` on algorithms settings
- `seed` on algorithms settings? Could that ever be searchable?

Reply:
I agree, but with the current hyperparams registration mechanism, these parameters would not be printed in the `__repr__(html__)` methods if they are defined as separate properties

Reply:
Nice catch, that would need to be handled manually as well.